server/controllers/transactions.js

```diff
@@ -18,97 +18,76 @@ const getOldestTransaction = async (walletUuid, tokenUuid) => {
 };
 
 module.exports = {
-  create({ body, params }, res) {
-    let data;
-    const { amount, fromAddress, toAddress, signed } = body;
-    // get Uint8array version of publicKey for verification
-    const verifyingWallet = StellarSdk.Keypair.fromPublicKey(fromAddress);
-    // remake the messageObj from params
-    const msg = dec(
-      JSON.stringify({
+  async create({ body, params }, res) {
```

Comment:
i'm thinking we should have a separate function for create vs 'share'

create will be generating a new challenge (tx) which will definitely always happen on chain (requires a sponsor address, but no from/to)

share will be generating a new transaction with a parent id and can happen off chain and requires a from/to

Reply:
I agree. I will begin implementation.